### PR TITLE
Fixed class method utils

### PR DESCRIPTION
**Pull request by matllubos**

### CODE REVIEW

File: chamber/utils/__init__.py [hunk 1]
```diff
@@ -43,15 +47,19 @@ def keep_spacing(value, autoescape=True):
...
-    if len(method_kwargs_names) == len(method_kwargs):
-        return method(**method_kwargs)
+def call_function_with_unknown_input(function, **kwargs):
+    """
+    Call function and use kwargs from input if function requires them.
+    :param function: function to call.
+    :param kwargs: function input kwargs or extra kwargs which will not be used.
+    :return: function result or raised InvalidFunctionArguments exception.
+    """
+    function_kwargs_names = set(inspect.signature(function).parameters.keys())
```
Comment (rubickcz) [inline chamber/utils/__init__.py:57]: Shouldn't you differentiate between kwargs with default value and kwargs without it? (kwargs with default value don't have to be specified in input)

Comment (matllubos) [reply chamber/utils/__init__.py:57]: Interesting idea. I will try it

Comment (matllubos) [reply chamber/utils/__init__.py:57]: done

**Review by rubickcz — APPROVED** [review]

File: chamber/utils/__init__.py [hunk 2]
```diff
@@ -43,15 +47,24 @@ def keep_spacing(value, autoescape=True):
...
+    )
 
-    if len(method_kwargs_names) == len(method_kwargs):
-        return method(**method_kwargs)
+    function_kwargs = {k: v for k, v in kwargs.items() if k in function_kwargs_names}
+    if required_function_kwargs_names <= set(function_kwargs.keys()):
+        return function(**function_kwargs)
     else:
-        raise RuntimeError('Invalid method parameters')
+        raise InvalidFunctionArguments('Function {} arguments has not subset of fun kwargs'.format(function))
```
Comment (Formulka) [inline chamber/utils/__init__.py:67]: Maybe a better message would be `Function {} missing required arguments {}` and include the missing required arguments?

**Pull request merged**